src/library/scala/collection/concurrent/TrieMap.scala

```diff
@@ -822,15 +824,15 @@ final class TrieMap[K, V] private (r: AnyRef, rtupd: AtomicReferenceFieldUpdater
     *
     * @param k the key to remove
     * @param v the value compare with the value found associated with the key
-    * @param removeAlways if true, then `k` will be removed whether or not its value matches `v`
-    *                     if false, then `k` will ONLY be removed if its value matches `v`
+    * @param removalPolicy policy deciding whether to remove `k` based on `v` and the
+   *                       current value associated with `k` (Always, FullEquals, or ReferenceEq)
     * @return an Option[V] indicating the previous value
     */
-  @tailrec private def removehc(k: K, v: V, removeAlways: Boolean, hc: Int): Option[V] = {
+  @tailrec private def removehc(k: K, v: V, removalPolicy: RemovalPolicy, hc: Int): Option[V] = {
```

Comment:
I thought this cleaner than adding a second `Boolean` parameter `fullEquals`. however, I don't know if it causes a performance hit. I'm not attached to this implementation over others. other options include 2 `Boolean` parameters, or an `Int` parameter and 3 constants with the same or similar names to the current `RemovalPolicy` names.

Reply:
Int constants (marked `final val`, so they more easily turn into field access) tend to be optimized quite reliably.  If you are going to benchmark different options, I'd go with whatever is clearest.  If you just want the most reliable path to fast code, I'd use the `Int`.  (Even that isn't certain, since the comparison might be better optimized by pulling it out.  But I've almost never found improvements over a single `Int` in this kind of scenario.)